demisto_sdk/dev_tools/spell_checker.py

```diff
@@ -0,0 +1,126 @@
+            with open(self.checked_file_path, 'r') as yaml_file:
+                yml_info = yaml.safe_load(yaml_file)
+
+            self.check_yaml(yml_info=yml_info)
+
+        else:
+            print_error("The file {} is not supported for spell-check command.\n"
+                        "Only .yml or .md files are supported.".format(self.checked_file_path))
+            return False
+
+        if self.unknown_words:
+            print_error(u"Words that might be misspelled were found in {}:\n\n{}".format(
+                self.checked_file_path, '\n'.join(self.unknown_words)))
+            return False
+
+        print_color("No misspelled words found in {}".format(self.checked_file_path), LOG_COLORS.GREEN)
+        return True
+
+    def check_md_file(self):
+        """Runs spell check on .md file. Adds unknown words to given unknown_words set.
+        """
+        with open(self.checked_file_path, 'r') as md_file:
+            md_file_lines = md_file.readlines()
+
+        for line in md_file_lines:
+            for word in line.split():
+                if word.isalpha() and self.spellchecker.unknown([word]):
+                    self.unknown_words.add(word)
+
+    def check_yaml(self, yml_info):
+        """Runs spell check on .yml file. Adds unknown words to given unknown_words set.
+
+        Args:
+            yml_info (Loader): A line list of the .md file contents
+        """
+        # yml file is parsed as a large dictionary. Each key in the large dictionary
+        # has a value which can be comprised of a string, a sub-dictionary or a list of dictionaries.
+        # The following code separates the spell check to these cases.
+        for key, value in yml_info.items():
+            # case 1: the value is a user visible string.
+            if key in USER_VISIBLE_LINE_KEYS:
+
+                # separate the string to individual words
+                for word in value.split():
+
+                    # Drop commas, full-stops and brackets.
+                    if word.endswith(',') or word.endswith('.') or word.endswith(')') or word.endswith(':'):
+                        word = word[:-1]
+
+                    if word.startswith('('):
+                        word = word[1:]
+
+                    # if a word is comprised of only letters (no punctuation ia checked!)
+                    if word.isalpha() and self.spellchecker.unknown([word]):
+                        self.unknown_words.add(word)
+
+            else:
+                # case 2: a sub-dictionary
+                if isinstance(value, dict):
+                    # ignore command arguments in playbooks - no need to check these.
+                    if key != 'scriptarguments':
+                        self.check_yaml(value)
+
+                # case 3: a list of dictionaries
+                elif isinstance(value, list):
+                    for sub_dict in value:
+                        if isinstance(sub_dict, dict):
```

Comment:
Are we sure all of the sub_dicts will be a dict and none will be a list?

Reply:
I parsed both an integration and a playbook.
In an integration, I didn't find '[[' - so no nested lists as far as I can tell.
In a playbook the only '[[' I found were for operators and thats under "scriptarguments" which we ignore - like so:
'**scriptarguments**': {'format': {}, 'thread-id': {'complex': {'root': 'Gmail', 'filters': **[[**{'operator': 'isEqualString', 'left': {'value': {'simple': 'Gmail.Mailbox'}, 'iscontext': True}, 'right': {'value': {'simple': 'admin@demistodev.com'}}}]]...